### PR TITLE
Migrate to MoviebaseApp/trakt-api

### PR DESCRIPTION
**Pull request by chrisnkrueger**
Initial try to integrate the trakt-api. There is some more work:

- [ ] Testing the trakt-api requests and reviewing the API
- [ ] Test the ZonedDateTime in Kotlin Datetime library (never looked into the parsing)
- [ ] Authentication / login with trakt
- [ ] Fix the unit tests
- [ ] Improve module dependency, currently trakt-auth -> trakt-api ->trakt-auth- android

Related to #1091

### CODE REVIEW

File: api/trakt-auth/api/src/main/java/app/tivi/trakt/TraktManager.kt [hunk 1]
```diff
@@ -36,7 +35,6 @@ import me.tatarka.inject.annotations.Inject
 class TraktManager(
     private val dispatchers: AppCoroutineDispatchers,
     private val showTasks: ShowTasks,
-    private val traktClient: Lazy<TraktV2>,
     private val authStore: AuthStore,
 ) {
```
Comment (chrisnkrueger) [inline api/trakt-auth/api/src/main/java/app/tivi/trakt/TraktManager.kt:39]: We aren't storing the bearer token in the client anymore. Furthermore, it should be delivered from AuthStore directly.

**Comment by chrisbanes** [comment]
Thanks for this!

> Test the ZonedDateTime in Kotlin Datetime library (never looked into the parsing)

kotlinx datetime doesn't contain a `ZonedDateTime` type. Luckily for us, all responses from Trakt sent in UTC ([here](https://trakt.docs.apiary.io/#introduction/dates)) so we can just use `Instant`

> Improve module dependency, currently trakt-auth -> trakt-api ->trakt-auth- android

Yeah, I need to fix this. It's been something I've wanted to fix for ages but low priority.

File: .idea/artifacts/tmdb_jvm.xml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 <component name="ArtifactManager">
   <artifact type="jar" name="tmdb-jvm">
     <output-path>$PROJECT_DIR$/api/tmdb/build/libs</output-path>
-    <root id="archive" name="tmdb-jvm.jar" />
+    <root id="archive" name="tmdb-jvm.jar">
```
[the remark this answers is not in the document]
Comment (chrisbanes) [reply .idea/artifacts/tmdb_jvm.xml:4]: I think this is fine. I should probably git rm and ignore these anyway.

**Comment by chrisbanes** [comment]
Closing this as I've moved the commits to #1156

**Pull request closed without merging**